src/middleware/options/RetryHandlerOptions.ts

```diff
@@ -87,7 +86,7 @@ export class RetryHandlerOptions implements MiddlewareOptions {
 	 * @param {ShouldRetry} [shouldRetry = RetryHandlerOptions.DEFAULT_SHOULD_RETRY] - The shouldRetry callback function
 	 * @returns An instance of RetryHandlerOptions
 	 */
-	public constructor(delay: number = RetryHandlerOptions.DEFAULT_DELAY, maxRetries: number = RetryHandlerOptions.DEFAULT_MAX_RETRIES, shouldRetry: ShouldRetry = RetryHandlerOptions.DEFAULT_SHOULD_RETRY) {
+	public constructor(delay: number = RetryHandlerOptions.DEFAULT_DELAY, maxRetries: number = RetryHandlerOptions.DEFAULT_MAX_RETRIES, shouldRetry: ShouldRetry = RetryHandlerOptions.defaultShouldRetry) {
```

Comment:
```suggestion
	public constructor(public delay: number = RetryHandlerOptions.DEFAULT_DELAY, public maxRetries: number = RetryHandlerOptions.DEFAULT_MAX_RETRIES, public shouldRetry: ShouldRetry = RetryHandlerOptions.defaultShouldRetry) {
```
and remove assignments below